pubsub.go

```diff
@@ -43,6 +46,19 @@ type ReceiveSettings struct {
 	Settings pubsub.ReceiveSettings
 }
 
+// SubscriptionSettings is an extension of Google PubSub's SubscriptionConfig that
+// enables further configuration for a subscription of a topic
+type SubscriptionSettings struct {
```

Comment:
based off yours conversation with @aaronluannguyen , if you are going to pass in configurations as params for `CreateSubscriptions` and `CreateTopics` i feel like it could be better and more aligned to stick with Google's PubSub naming and change them from `SubscriptionSettings` -> `SubscriptionConfig` and  `TopicSettings` -> `TopicConfig`, as opposed to `PublishSettings` and `ReceiveSettings` which still stay at the global level

Reply:
Yeah, I agree. If we're moving towards passing in settings/config per call of these methods, then I like calling it `Config` vs `Settings`